internal/pkg/containers/cri/cri.go

```diff
@@ -225,6 +227,18 @@ func (i *inspector) buildPod(sandbox *runtimeapi.PodSandbox) (*ctrs.Pod, error)
 					pod.Containers[0].Digest = digest
 				}
 			}
+
+			if sandboxStatus.Linux.Namespaces.Options.Network == runtimeapi.NamespaceMode_POD {
+				if ns, ok := verboseInfo["runtimeSpec"].(map[string]interface{})["linux"].(map[string]interface{})["namespaces"].([]interface{}); ok {
```

Comment:
would `runtimeSpec.linux` be ever not set?

Reply:
Line 231 is logic which is not needed, and only there to skip unnecessary code.
I am happy to remove `if sandboxStatus.Linux.Namespaces.Options.Network` it is only there as on 233 starts a loop which will be pointless if the container is not network namespaced.

If your question is about the `if` on 232 itself, as this is unmarshaled json any assumption on how the json should look like could lead to a nil pointer reference in the future. (which is of course not worth stripping the if, also that why surrounding code like `verboseInfo["images")` and `verboseInfo["pod"]` have it.

Please let me know if I should remove it.

Reply:
I modified this code to be less "panicky" to check for every layer in JSON